app-servers.tf

```diff
@@ -1,17 +1,60 @@
-/* App servers */
+/* API and Docker servers */
 resource "aws_instance" "tsuru-app" {
-  count = 3
+  count = 2
   ami = "${lookup(var.amis, var.region)}"
   instance_type = "t2.medium"
   subnet_id = "${aws_subnet.private1.id}"
   security_groups = ["${aws_security_group.default.id}"]
   key_name = "${aws_key_pair.deployer.key_name}"
   user_data = "${file(\"cloud-config/app.yml\")}"
-  tags = { 
+  tags = {
     Name = "tsuru-app-${count.index}"
   }
 }
 
+/* Gandalf server */
+resource "aws_instance" "gandalf" {
+  ami = "${lookup(var.amis, var.region)}"
+  instance_type = "t2.medium"
+  subnet_id = "${aws_subnet.public1.id}"
+  associate_public_ip_address = true
+  security_groups = ["${aws_security_group.default.id}", "${aws_security_group.gandalf.id}"]
+  key_name = "${aws_key_pair.deployer.key_name}"
+  user_data = "${file(\"cloud-config/app.yml\")}"
+  tags = {
+    Name = "tsuru-app-gandalf"
+  }
+}
+
+/* API external Load balancer */
+resource "aws_elb" "api-ext" {
+  name = "tsuru-api-elb-ext"
+  subnets = ["${aws_subnet.public1.id}", "${aws_subnet.public2.id}"]
+  instances = [ "${aws_instance.tsuru-app.0.id}" ]
+  security_groups = ["${aws_security_group.default.id}", "${aws_security_group.web.id}"]
+  listener {
+    instance_port = 8080
+    instance_protocol = "http"
+    lb_port = 8080
+    lb_protocol = "http"
+  }
+}
+
+/* API internal Load balancer */
+resource "aws_elb" "api-int" {
```

Comment:
Nassim has helped answer this. It was created because it was thought that some internal components needed to talk to the API and they couldn't/shouldn't (?) go through the public facing ELB.

As I understand, by using the hostname's that AWS provides, split-horizon DNS will resolve to the appropriate external or internal address. Which means that security groups work as expected and you don't get charge for the traffic.

Nevertheless, it was later determined that the internal components don't call the API and that we can refactor these two ELBs out. That should be done before merging this PR.